sql-plugin/src/main/spark320/scala/com/nvidia/spark/rapids/shims/HybridFileSourceScanExecMeta.scala

```diff
@@ -0,0 +1,216 @@
+  // if possible. Instead regarding filters as childExprs of current Meta, we create
+  // a new meta for SubqueryBroadcastExec. The reason is that the native replacement of
+  // FileSourceScan is independent from the replacement of the partitionFilters.
+  private lazy val partitionFilters = {
+    val convertBroadcast = (bc: SubqueryBroadcastExec) => {
+      val meta = GpuOverrides.wrapAndTagPlan(bc, conf)
+      meta.tagForExplain()
+      meta.convertIfNeeded().asInstanceOf[BaseSubqueryExec]
+    }
+    wrapped.partitionFilters.map { filter =>
+      filter.transformDown {
+        case dpe@DynamicPruningExpression(inSub: InSubqueryExec) =>
+          inSub.plan match {
+            case bc: SubqueryBroadcastExec =>
+              dpe.copy(inSub.copy(plan = convertBroadcast(bc)))
+            case reuse@ReusedSubqueryExec(bc: SubqueryBroadcastExec) =>
+              dpe.copy(inSub.copy(plan = reuse.copy(convertBroadcast(bc))))
+            case _ =>
+              dpe
+          }
+      }
+    }
+  }
+
+  // partition filters and data filters are not run on the GPU
+  override val childExprs: Seq[ExprMeta[_]] = Seq.empty
+
+  override def tagPlanForGpu(): Unit = {
+    val cls = wrapped.relation.fileFormat.getClass
+    if (cls != classOf[ParquetFileFormat]) {
+      willNotWorkOnGpu(s"unsupported file format: ${cls.getCanonicalName}")
+    }
+  }
+
+  override def convertToGpu(): GpuExec = {
+    // Modifies the original plan to support DPP
+    val fixedExec = wrapped.copy(partitionFilters = partitionFilters)
+    HybridFileSourceScanExec(fixedExec)(conf)
+  }
+}
+
+object HybridFileSourceScanExecMeta {
+  private val HYBRID_JAR_PLUGIN_CLASS_NAME = "com.nvidia.spark.rapids.hybrid.HybridPluginWrapper"
+
+  // Determines whether using HybridScan or GpuScan
+  def useHybridScan(conf: RapidsConf, fsse: FileSourceScanExec): Boolean = {
+    val isEnabled = if (conf.useHybridParquetReader) {
+      require(conf.loadHybridBackend,
+        "Hybrid backend was NOT loaded during the launch of spark-rapids plugin")
+      true
+    } else {
+      false
+    }
+    // For the time being, only support reading Parquet
+    lazy val isParquet = fsse.relation.fileFormat.getClass == classOf[ParquetFileFormat]
+    // Check if data types of all fields are supported by HybridParquetReader
+    lazy val allSupportedTypes = fsse.requiredSchema.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, {
+        // For the time being, the native backend may return incorrect results over nestedMap
+        case MapType(kt, vt, _) if kt.isInstanceOf[MapType] || vt.isInstanceOf[MapType] => false
+        // For the time being, BinaryType is not supported yet
+        case _: BinaryType => false
+        case _ => true
+      })
+    }
+    // TODO: supports BucketedScan
+    lazy val noBucketedScan = !fsse.bucketedScan
+    // TODO: supports working along with Alluxio
```

Comment:
We don't need to do that. The Alluxio work is essentially dead.

https://github.com/NVIDIA/spark-rapids/issues/9332#issuecomment-1777952590

Reply:
Removed

Reply:
Removed the TODO: 
// TODO: supports working along with Alluxio